src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs

```diff
@@ -730,6 +731,83 @@ public async Task PostImportAsync()
+
+            if (blittableJson.TryGet(nameof(DatabaseSmugglerOptions.OperateOnDatabaseRecordTypes), out string operateOnDatabaseRecordTypes) == false &&
+                blittableJson.TryGet(nameof(DatabaseSmugglerOptions.OperateOnDatabaseRecordTypes), out int _) == false)
+                throw new InvalidOperationException($"Invalid value for {nameof(DatabaseSmugglerOptions.OperateOnDatabaseRecordTypes)}, must be non-empty 'string' or 'int'");
+
+            DynamicJsonValue djv = null;
+            if (operateOnTypes != null && Enum.TryParse(typeof(DatabaseItemType), operateOnTypes, out _) == false)
+            {
+                CheckClientVersion();
+
+                var itemsTypes = DatabaseItemType.None;
+                var types = operateOnTypes.Split(", ");
+
+                foreach (var type in types)
+                {
+                    if (Enum.TryParse(typeof(DatabaseItemType), type, true, out var result) == false || result is DatabaseItemType existing == false)
+                        continue;
+
+                    itemsTypes |= existing;
+                }
+
+                djv = new DynamicJsonValue(blittableJson)
+                {
+                    [nameof(DatabaseSmugglerOptions.OperateOnTypes)] = itemsTypes
+                };
+            }
+
+            if (operateOnDatabaseRecordTypes != null && Enum.TryParse(typeof(DatabaseRecordItemType), operateOnDatabaseRecordTypes, out _) == false)
+            {
+                CheckClientVersion();
+
+                var databaseRecordItemsTypes = DatabaseRecordItemType.None;
+                var types = operateOnDatabaseRecordTypes.Split(", ");
+
+                foreach (var type in types)
+                {
+                    if (Enum.TryParse(typeof(DatabaseRecordItemType), type, true, out var result) == false || result is DatabaseRecordItemType existing == false)
+                        continue;
+
+                    databaseRecordItemsTypes |= existing;
+                }
+
+                if (djv == null)
+                {
+                    djv = new DynamicJsonValue(blittableJson)
+                    {
+                        [nameof(DatabaseSmugglerOptions.OperateOnDatabaseRecordTypes)] = databaseRecordItemsTypes
+                    };
+                }
+                else
+                {
+                    djv[nameof(DatabaseSmugglerOptions.OperateOnDatabaseRecordTypes)] = databaseRecordItemsTypes;
+                }
+            }
+
+            if (djv == null)
+                return;
+
+            blittableJson.Modifications = djv;
+            using (var old = blittableJson)
+            {
+                blittableJson = context.ReadObject(blittableJson, "item-types");
+            }
+        }
+
+        private void CheckClientVersion()
+        {
+            if (RequestRouter.TryGetClientVersion(HttpContext, out var version) == false || version.Build <= RavenVersionAttribute.Instance.BuildVersion)
```

Comment:
This exception says 0 about the actual error. It actually hides the real issue.

Reply:
done (I think?)